src/klein/test/test_resource.py

```diff
@@ -100,8 +101,8 @@ def write(data):
                 "Request.finish was called."
             )
 
-    def getWrittenData():
-        return request._written.getvalue()
+    def getWrittenData() -> bytes:
+        return cast(BytesIO, request._written).getvalue()
```

Comment:
I think the issue here is that `_written` is sloppily hung off of `request` here so mypy doesn't know it's type.

#476 changes this function to a class, which mypy likes better.

Reply:
Yeah, this kind of ad-hoc nonsense is something that I'm glad Mypy is forcing us to stop doing.